### PR TITLE
Use 'webpack-merge' for dev/prod builds.

### LINKED ISSUE #100: ocelot-ide.org is running the development build of React
I spotted this while using the "React Developer Tools" plugin. Is this something worth mentioning because I think the production build enhances the performance of the page?

### PR DESCRIPTION
**Pull request by sp1tz**
Fix #100.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -41,18 +41,19 @@
...
+    "webpack-merge": "^4.2.1"
   },
   "scripts": {
-    "gen-version": "echo \"let OCELOTVERSION = '$(git rev-parse --short HEAD)';\nexport { OCELOTVERSION };\" > src/version.ts",
-    "build": "yarn run gen-version && webpack --progress",
-    "start": "yarn run gen-version && webpack-dev-server --open",
+    "gen-version": "echo \"export const OCELOTVERSION = '$(git rev-parse --short HEAD)';\" > src/version.ts",
+    "build": "yarn gen-version && webpack --progress --config webpack.prod.js",
+    "start": "yarn gen-version && webpack-dev-server --open --config webpack.dev.js",
     "serve-local": "http-server build",
```
Comment (sp1tz) [inline frontend/package.json]: I suggest we remove this. Any reason not to, given we already have the webpack dev-server?

**Comment by arjunguha** [comment]
Should we really use prod mode? The dev mode seems fast enough, and if it is easier to debug, that's great.

**Comment by sp1tz** [comment]
I think we should as it's the recommended configuration from [webpack](https://webpack.js.org/guides/production/) themselves. Production will still have source-maps (just not inline-source-maps); though the vast majority of debugging will be done in development mode (_i.e._ not on the actual deployed site).

**Comment by arjunguha** [comment]
I know they recommend that. But I don’t think their rationale matters for us.

**Comment by sp1tz** [comment]
It's not going to make much of a difference in overall app performance, but at the same time we'd not be making any sacrifices. The `devtool` property is set to `source-map` currently, as it would be in the production configuration (_i.e._ for debugging).

**Comment by arjunguha** [comment]
So there is no reason to do this. The only cost is a more complex build system. The build is alrady complex enough. Let's not do this.

**Pull request closed without merging**